luigi_pipeline/lib/model/gcnv_mt_schema.py

```diff
@@ -0,0 +1,210 @@
+import datetime
+import re
+
+import hail as hl
+
+from lib.model.base_mt_schema import BaseMTSchema, row_annotation
+from lib.model.seqr_mt_schema import SeqrGenotypesSchema, SeqrVariantsAndGenotypesSchema
+
+from hail_scripts.computed_fields import variant_id
+
+GENE_ID = "gene_id"
+MAJOR_CONSEQUENCE = "major_consequence"
+
+def parse_genes(gene_col: hl.expr.StringExpression) -> hl.expr.SetExpression:
+    """
+    Convert a string-ified gene list to a set()
+    """
+    return gene_col.split(',').filter(
+        lambda gene: ~hl.set({'None', 'null', 'NA', ''}).contains(gene)
+    ).map(
+        lambda gene: gene.split(r'\.')[0]
+    )
+
+def hl_agg_collect_set_union(gene_col: hl.expr.SetExpression) -> hl.expr.SetExpression:
+    """
+    aggregate with the set union operator
+    """
+
+    return hl.set(
+        hl.flatten(
+            hl.agg.collect(gene_col)
```

Comment:
would ` hl.flatten(hl.agg.collect_as_set` work?

Reply:
As is... `hl.flatten(hl.agg.collect_as_set` throws an error for me (I'm not fully sure why...)

I think we would actually still need the outer `hl.set` as well as an inner `hl.array` cast to make that work.  

For example:

```
t1 = hl.Table.parallelize(
    ...:      [
    ...:          {"variant": ["1", "2"], "sv_type":"2", "sample_fix": "1", "data" : 5},
    ...:          {"variant": ["1", "4"], "sv_type":"2", "sample_fix": "2", "data" : 6},
    ...:          {"variant": ["2", "5", "4"], "sv_type":"2", "sample_fix": "3", "data" : 7},
    ...:          {"variant": ["1", "3", "4"], "sv_type":"2", "sample_fix": "4", "data" : 8},
    ...:      ],
    ...:      hl.tstruct(variant=hl.dtype('array<str>'), sv_type=hl.dtype('str'), sample_fix=hl.dtype('str'), data=hl.dtype('int32')),
    ...:  )
```

```
In [14]: t1.aggregate(hl.flatten(hl.array(hl.agg.collect_as_set(t1.variant))))
Out[14]: ['1', '2', '1', '3', '4', '1', '4', '2', '5', '4']

In [15]: t1.aggregate(hl.set(hl.flatten(hl.array(hl.agg.collect_as_set(t1.variant)))))
Out[15]: {'1', '2', '3', '4', '5'}
```

versus

```
In [12]: t1.aggregate(hl.flatten(hl.agg.collect(t1.variant)))
Out[12]: ['1', '2', '1', '4', '2', '5', '4', '1', '3', '4']
In [12]: t1.aggregate(hl.flatten(hl.agg.collect(t1.variant)))
Out[12]: ['1', '2', '1', '4', '2', '5', '4', '1', '3', '4']

In [13]: t1.aggregate(hl.set(hl.flatten(hl.agg.collect(t1.variant))))
Out[13]: {'1', '2', '3', '4', '5'}
```

Reply:
its weird that its throwing an error for you - the documentation for `hl.flatten` explicitly states that  it can accept either a SetExpression or an ArrayExpression

Reply:
> this can work if I go back to having the parsed genes be sets instead of an array.

I think thats better in some cases, and you can always cast it back to an array if you really need to in one or two places

Reply:
👍 and fixed!